sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateSchemaCompatibilityChecker.scala

```diff
@@ -34,6 +35,7 @@ class StateSchemaCompatibilityChecker(
   private val storeCpLocation = providerId.storeId.storeCheckpointLocation()
   private val fm = CheckpointFileManager.create(storeCpLocation, hadoopConf)
   private val schemaFileLocation = schemaFile(storeCpLocation)
+  private val schemaWriter = SchemaWriter.createSchemaWriter("v2")
```

Comment:
Let's make this simple to pass StateSchemaCompatibilityChecker.VERSION instead. Doesn't need to let this class know how the version string looks like.